wfc_gh/WaveFunctionCollapse.cs

```diff
@@ -463,27 +463,25 @@ protected override void SolveInstance(IGH_DataAccess DA)
 
             bool foundDeterministic = false;
             uint attempts = 0;
-
             uint maxObservations = UInt32.MaxValue;
             uint spentObservations = 0;
             
             unsafe{
-                while (!foundDeterministic && attempts < maxAttempts)
+                while (true)
                 {
                     var result = Native.wfc_observe(wfcWorldStateHandle,
                                                     wfcRngStateHandle,
-                                                    (UIntPtr)maxObservations,
+                                                    axObservations,
                                                     &spentObservations);
-                    if (result == WfcObserveResult.Deterministic)
-                    {
-                        foundDeterministic = true;
-                    }
-                    else
-                    {
-                        Native.wfc_world_state_clone_from(wfcWorldStateHandle,
-                                                        wfcWorldStateHandleBackup);
-                    }
                     attempts++;
+                    foundDeterministic = observationResult == WfcObserveResult.Deterministic;
+
+                    if (foundDeterministic
+                        || observationResult == WfcObserveResult.Nondeterministic
+                        || attempts == maxAttempts) {
+                        break;
+                    }
```

Comment:
So this now breaks for all deterministic, nondeterministic, or if attempts are exhausted? That doesn't seem right. You probably wanted to use && between the last two equalities in the condition.

Reply:
I believe it is correct. Let's break it down:

1. If the solution is `contradictory`, then it is bad in all cases and there should be another attempt... 
2. ...unless the `maximum number of attempts` has been exhausted. Then it should return any solution, even `contradictory`.
3. The `deterministic` solution is always good and should be returned.
4. The `non-deterministic` solution should not happen for solver with unlimited observations - either it finds a `deterministic` or `contradictory` solution or runs out of observations - which is set to `uint32::MAX`. The number of observations cannot be higher than the count of slots. The maximum is so high, that it is virtually impossible to reach. As we do not pretend there is any "unlimited number of observations" but rather a very high limit, then even this virtually impossible case is valid.
5. For limited number of observations it is valid to return a `non-deterministic` result even if the `maxAttempts` has not been reached. Your suggestion prevents this option.